api/db/changelog/db.changelog-157-add-user-access-tier.xml

```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+  xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                      http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+  <changeSet author="thibault" id="changelog-157-add-user-access-tier">
+
+    <createTable tableName="user_access_tier">
+      <column name="user_access_tier_id" type="bigint" autoIncrement="true">
+        <constraints primaryKey="true" nullable="false"/>
+      </column>
+      <column name="user_id" type="bigint">
+        <constraints nullable="false"
+          foreignKeyName="fk_user_access_tier_user"
+          references="user(user_id)"
+          deleteCascade="true"/>
+      </column>
+      <column name="access_tier_id" type="bigint">
+        <constraints nullable="false"
+          foreignKeyName="fk_user_access_tier_access_tier"
+          references="access_tier(access_tier_id)"
+          deleteCascade="true"/>
+      </column>
+      <column name="access_status" type="tinyint">
+        <constraints nullable="false"/>
+      </column>
+      <column name="first_enabled" type="datetime">
+       <constraints nullable="false"/>
+      </column>
+      <column name="last_updated" type="datetime">
+        <constraints nullable="false"/>
+      </column>
+    </createTable>
+
+    <addUniqueConstraint
+      columnNames="user_id, access_tier_id"
+      constraintName="user_access_tier_pair"
+      tableName="user_access_tier"
+    />
+
+    <!--
+    migration from user.data_access_level:
+    Populate this table with a Registered Tier entry for all users who are Registered.
+    There are no Protected or Controlled Tier users to migrate.
+    -->
+
+    <sql>
+      INSERT INTO user_access_tier
+      (user_id, access_tier_id, access_status, first_enabled, last_updated)
+
+      SELECT
+      u.user_id,
+      a.access_tier_id,
+      1 AS access_status, -- TierAccessStatus.ENABLED
```

Comment:
What is the semantic difference between a user with a DISABLED tier assocation vs one without a row at all?

Reply:
When determining current access, they are equivalent.  DISABLED means a user had tier access at one point, but now does not.  This lets us:
* distinguish between never-had-access and currently-doesn't-have
* keep a record of the first time a user qualified for the tier (a la firstUserRegistrationTime)

I don't know whether either of these will prove to be useful.  But it's easy to do now and would be hard to add later.